---
- name: Install gluster
  package:
    name: glusterfs-server
    state: present

- name: Start the service
  service:
    name: glusterd
    enabled: True
    state: started

- include_tasks: firewalld.yml
  when: use_firewalld|bool

- name: Create the brick
  file:
    name: "{{ brick_path }}"
    state: directory
    
- name: Create the shared volume
  gluster_volume:
    state: present
    name: "{{ volume_name }}"
    bricks: "{{ brick_path }}"
    # TODO fix me later by having a separate disk
    force: True
    replicas: 2
    cluster: "{{ groups[group]|map('extract', hostvars, ['ansible_' + interface, 'ipv4', 'address']) | list  }}"
  run_once: true
  # for some reason, this module fail if the cluster is running
  ignore_errors: True

- name: Create the mount
  file:
    name: "{{ mount_path }}"
    state: directory

- name: Mount Gluster
  mount:
    fstype: glusterfs
    src: "{{ hostvars[inventory_hostname]['ansible_' + interface]['ipv4']['address'] }}:{{ volume_name }}"
    path: "{{ mount_path }}"
    state: mounted

# TODO take care of letsencrypt setup
# - link /etc/letsencrypt to /srv
# - cronjob that run only if the keepalive is master
# - likely selinux permission
